torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -549,6 +564,59 @@ void RequestCallbackImpl::processRpc(
       markComplete(std::move(CleanupAutogradContextResp()).toMessage());
       return;
     }
+    case MessageType::RUN_WITH_PROFILING_REQ: {
+      auto& rpcWithProfilingReq = static_cast<RpcWithProfilingReq&>(rpc);
+      auto wrappedMsgType = rpcWithProfilingReq.wrappedMessageType();
+      const auto profilingConfig = rpcWithProfilingReq.getProfilingConfig();
+      auto wrappedRpcResponseFuture = std::make_shared<FutureMessage>();
+      auto fromWorkerId = rpcWithProfilingReq.fromWorkerId();
+      // Enable the profiler with the config from the sender.
+      std::vector<torch::autograd::profiler::Event> profiledEvents;
+      {
+        torch::autograd::profiler::TLSProfilerGuard g(
+            profilingConfig,
+            [&profiledEvents](
+                // NOLINTNEXTLINE
+                std::vector<std::vector<torch::autograd::profiler::Event>>
+                    event_lists) {
```

Comment:
Since we are using `NOLINTNEXTLINE`, I assume the intention is to not using const-ref? Any reason for this? And if this cannot be a reference, shall we move and emplace_back them into the `profiledEvents`? (or move_iterator + insert)?

Reply:
I was running into some issues that indicated fields in the `Event`s here were being overwritten by other memory before I used them, so I changed it to use a copy, although, I did change some things after. Let me look into this again.

Reply:
Should be update to use constref in the next update